lib/mongo/auth/scram/conversation.rb

```diff
@@ -429,9 +429,15 @@ def xor(first, second)
           first.bytes.zip(second.bytes).map{ |(a,b)| (a ^ b).chr }.join('')
         end
 
+        def secure_compare(a, b)
+          check = a.bytesize ^ b.bytesize
+          a.bytes.zip(b.bytes){ |x, y| check |= x ^ y.to_i }
```

Comment:
If they lengths are not equal then null bytes are used to pad the shorter to the length of the longer.